### PR TITLE
Fix `--showConfig` to show transitively implied options that vary from the default config

### LINKED ISSUE #59442: Potentially wrong 'showConfig' results for default values, as it's used in the checker differently.
### 🔎 Search Terms

"showConfig result moduleResolution wrong"

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about showConfig

### ⏯ Playground Link

_No response_

### 💻 Code

tsconfig:
```json
{
  "compilerOptions": {
  }
}
```

cli call:
`tsc --showConfig`

### 🙁 Actual behavior

```json
{    
  "compilerOptions": {}
}
```

### Wanted Output from `tsc --showConfig`

```json
{
  "compilerOptions": {
    "moduleResolution": "Node10",
    // And All other fields which have default values
  }
}
```

### Explanation
When we check utilities.ts: 8859 `computedOptions` we can see, that this moduleResolution has the following `computeValue` functionality and that its not the default value because it depends on the computed module value which is, if not set, `commonjs`. 
```typescript
{
dependencies: ["module", "target"],
  computeValue: (compilerOptions): ModuleResolutionKind => {
  let moduleResolution = compilerOptions.moduleResolution;
  if (moduleResolution === undefined) {
    switch (computedOptions.module.computeValue(compilerOptions)) {
      case ModuleKind.CommonJS:
        moduleResolution = ModuleResolutionKind.Node10;
        break;
      case ModuleKind.Node16:
        moduleResolution = ModuleResolutionKind.Node16;
        break;
      case ModuleKind.NodeNext:
        moduleResolution = ModuleResolutionKind.NodeNext;
        break;
      case ModuleKind.Preserve:
        moduleResolution = ModuleResolutionKind.Bundler;
        break;
      default:
        moduleResolution = ModuleResolutionKind.Classic;
        break;
    }
  }
  return moduleResolution;
},
    }
```

And in the compiler, it's used like this: 
```typescript
export const getEmitModuleResolutionKind = computedOptions.moduleResolution.computeValue;
// Usage:
getEmitModuleResolutionKind(compilerOptions)
```

So there is no checking for if the `dependencies` are used. So this means that the default value will be used even if there is no dependency specified in the tsconfig. 

But in the showConfig code (commandLineParser.ts: 2565) we can see, that it checks if some of it's dependencies are there:
```typescript
if (!providedKeys.has(option) && some(computedOptions[option as keyof typeof computedOptions].dependencies, dep => providedKeys.has(dep))) {
```

### Suggestion 
So I think the showConfig does not show the "real" values which are used during the execution of the code and I would suggest to change that to show all the properties if they have a value which is other than undefined.

### PR DESCRIPTION
**Pull request by andrewbranch**
Fixes #59442

#56701 made it so `--showConfig` shows implied compiler options if their implied value differs from what you’d get with no config. However, it mistakenly only showed implied options that were affected by an explicitly set option, while failing to show implied options that were affected by other implied options. For example, setting `--module nodenext` implies `--moduleResolution nodenext`, which implies `--resolvePackageJsonExports`. But `--resolvePackageJsonExports` previously didn’t show up with `--module nodenext --showConfig`.

### CODE REVIEW

File: src/compiler/commandLineParser.ts [hunk 1]
```diff
@@ -2641,6 +2642,18 @@ export function convertToTSConfig(configParseResult: ParsedCommandLine, configFi
     return config;
 }
 
+function optionDependsOn(option: string, dependsOn: Set<string>): boolean {
+    const seen = new Map<string, true>();
```
Comment (jakebailey) [inline src/compiler/commandLineParser.ts:2646]: Should this just be a Set?

(I'm guessing addToSeen is leftover from pre es6)

Comment (jakebailey) [reply src/compiler/commandLineParser.ts:2646]: I guess it has some handy behavior)

Comment (andrewbranch) [reply src/compiler/commandLineParser.ts:2646]: This has bugged me for a while, maybe I should update `addToSeen`

**Review by jakebailey — APPROVED** [review]
In any case, LGTM, the `seen` thing is a cleanup

**Review by jakebailey — APPROVED** [review]

**Pull request merged**